### ISSUE TITLE
LogExtension colorizes output incorrectly

### ISSUE THREAD

**Issue by kcamp** [cake-build/cake#1548]
### What You Are Seeing
The new `LogExtension` overloads for `(string)` and `(object)` break the token paletting

### What is Expected?
The console output to be colorized consistent with 0.17 and lower.

### What version of Cake are you using?
0.19.1

### Are you running on a 32 or 64 bit system?
64

### What environment are you running on?  Windows? Linux? Mac?
Windows

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)
Use build script content such as:
```
_context.Information("  Updating assembly [FileVersion] metadata...");
_context.Information("========================================");
_context.Information("  Updating from {0} to {1}...", oldVersion, newVersion);
_context.Information("  Successs!");
```

Because of the way the methods were overloaded/cascaded, the console output now colorizes the `string` or `object` parameters as though they were formatting arguments.  In the left-hand capture, you can see that the entire line is colorized with forecolor/backcolor instead of just the replacements.

![image](https://cloud.githubusercontent.com/assets/649593/24419743/34a526ae-13be-11e7-9dcc-945d6a6c6e8f.png)

**Comment by kcamp** [comment]
I think the easiest solution is to simply change the implementation for the new `string` and `object` overloads for the Debug/Verbose/Warning/Information/Error methods.

```
public static void Debug(this ICakeLog log, object value)
{
-  log.Debug("{0}", value);
+  log.Debug(string.Format("{0}", value));   // call the (string) overload
}

public static void Debug(this ICakeLog log, string value)
{
-  log.Debug("{0}", value);
+  log.Debug(value, null);  // cascade this call into the (string, params oject[]) overload
}
```

I have created a local build and this seems to be a reasonable solution.  The unit tests would need to be updated to meet the new expectations, as they currently fail.  Thoughts?  I can take on that work if it seems like a reasonable approach.

**Comment by devlead** [comment]
@kcamp the overloads were added because issues with sending strings  with special characters, many people log generated code and json which fails if sent.
So they were added specifically if you only sent a string it wouldn't be passed to the tokenizer.
So this behavior is per design so object/strings wouldn't break logging which they did on several occasions  pre v0.18.0, and they were added in a way to not break ABI.

**Comment by kcamp** [comment]
@devlead 
Your explanation makes sense; I understood it to be an intentional change.  The net effect, however, is that the entire line is treated as a token, and the coloration of those lines lose visual distinction from the lines that _do_ contain substitution.  I think the readability of the log suffers for it, and it's a departure from the prior versions.

As a solution that really addresses the colorization (instead of the actual formatting string), I've put together a much simpler fix that returns the console output to be more consistent with the versions prior to 0.18.  See 9063db02e for this example.  If that seems like a reasonable solution to the degradation in formatting, I can put together some tests and issue a PR against the same.

Thanks!

**Comment by devlead** [comment]
Fixed by #1864
